File: SNS24.API/DTOs/Users/UserRegisterDto.cs
using SNS24.WebApi.Models;
using SNS24.WebApi.Models.Common;

namespace SNS24.Api.DTOs.Users;

public class UserRegisterDto
{
    public string Email { get; set; }
    public string Password { get; set; }
    public string Username { get; set; }
    public DateTime BirthDate { get; set; }
    public string DocumentNumber { get; set; }
    public string PhoneNumber { get; set; }
    public string Gender { get; set; }
    public Address Address { get; set; }
    public string Name { get; set; }

    public int Role { get; set; } 
    
    // needed for doctor 
    public string? LicenseNumber { get; set; }
    public string? Specialty { get; set; }
    public IEnumerable<Institution>? Institutions { get; set; } = new HashSet<Institution>();
    
    // needed for patient
    public string? SNSNumber { get; set; }
}

File: webapp/src/app/shared/interfaces/auth/userRegisterDto.ts
import { Address } from "../user/address";
import { Institution } from "../institution";

export interface UserRegisterDto {
  email: string;
  password: string;
  username: string;
  birthDate: string;
  documentNumber: string;
  phoneNumber : string;
  gender: string;
  address: Address;
  name: string;
  role: number;
  licenseNumber?: string;
  specialty?: string;
  institutions?: Institution[];
  snsNumber?: string;
}
